src/main/java/com/google/cloud/tools/maven/DeployAllMojo.java

```diff
@@ -38,17 +38,51 @@ public void execute() throws MojoExecutionException, MojoFailureException {
       deployables.clear();
     }
 
-    String[] validYamls = {
-      "app.yaml", "cron.yaml", "dispatch.yaml", "dos.yaml", "index.yaml", "queue.yaml"
-    };
-    for (String yamlName : validYamls) {
-      File yaml = stagingDirectory.toPath().resolve(yamlName).toFile();
-      if (yaml.exists()) {
-        getLog().info("deployAll: Preparing to deploy " + yamlName);
-        deployables.add(yaml);
+    String[] configYamls = {"cron.yaml", "dispatch.yaml", "dos.yaml", "index.yaml", "queue.yaml"};
+    File appYaml = stagingDirectory.toPath().resolve("app.yaml").toFile();
+    if (appYaml.exists()) {
+      addDeployable(appYaml);
+    }
+
+    if (isStandardStaging()) {
+      // Check staging directory for app.yaml
+      if (!appYaml.exists()) {
+        throw new MojoExecutionException("Failed to deploy all: app.yaml not found.");
```

Comment:
nit: since the exception here is the same as the exception for flexible, this could probably be split up into two stages, one for deploying app.yaml, and one for deploying the others, such that this code doesn't have to be duplicated.